### PR TITLE
Adding Anuga

### PR DESCRIPTION
**Pull request by stoiver**
@conda-forge/help-python-c this package needs to use the mingw compiler on windows. I tried to setup the build section of the meta.yaml to use m2w64-toolchain and libpython on windows. Hope it works. 

Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by stoiver** [comment]
@conda-forge/help-python-c This recipe should be ready for review.

File: recipes/anuga/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/anuga-{{ version }}.tar.gz
+  sha256: {{ sha }}
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  skip: True  # [py<3.6]
+
+requirements:
```
Comment (carterbox) [inline recipes/anuga/meta.yaml:18]: It looks like there are too many dependencies in each requirements sub-section.

The build section is for build tools like compilers, cmake, etc. These packages must match the host platform when cross compiling.

The host section is for compiled libraries and python build requirements like python, setuptools, libpython, sometimes numpy (if compiled against their c API).

The run section is for packages only needed at runtime like scipy, matplotlib. Cython is rarely needed here.

Packages only needed to run the tests, go in a separate test/requires section. i.e. pytest

File: recipes/anuga/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    - setuptools
+    - utm
+    - wheel
+    - pip
+  run:
+    - python
+    - m2w64-gcc-libs  # [win]
+    - libpython       # [win]
+    - libstdcxx-ng    # [linux]
+    - libgcc-ng       # [linux]
```
Comment (carterbox) [inline recipes/anuga/meta.yaml:57]:
```suggestion
```
These are added automatically as run_exports from the compilers, and many times the cxx standard library isn't even used.

File: recipes/anuga/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    - Pmw
+    - pymetis
+    - pytest
+    - pytz
+    - scipy
+    - setuptools
+    - utm
+    - wheel
+    - pip
+  run:
```
Comment (carterbox) [inline recipes/anuga/meta.yaml:52]: These run dependencies should have version specifications that match the metadata in the setup.py.

**Review by carterbox — CHANGES_REQUESTED** [review]
This package is using an uncommon layout and build system. Thanks for you patience in advance.

File: recipes/anuga/meta.yaml (hunk 1 above)
Comment (stoiver) [reply recipes/anuga/meta.yaml:18]: @carterbox  thanks. I have tried to minimize the requirements as suggested. Just pushed changes so hopefully I have left all the requirements.

File: recipes/anuga/meta.yaml (hunk 2 above)
Comment (stoiver) [reply recipes/anuga/meta.yaml]: @carterbox  removed as suggested.

File: recipes/anuga/meta.yaml (hunk 3 above)
Comment (stoiver) [reply recipes/anuga/meta.yaml:30]: @carterbox I ended up removing these from host requirements.

**Comment by stoiver** [comment]
@conda-forge/help-python-c This recipe should be ready for review.

File: recipes/anuga/meta.yaml (hunk 3 above)
Comment (carterbox) [reply recipes/anuga/meta.yaml:32]: The packages in requirements/run should have version specifications which match upstream.

https://github.com/anuga-community/anuga_core/blob/c657e1e41900bcf47f568b2b9bf75892bb46430a/setup.py#L51-L71

File: recipes/anuga/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    - python
+    - _openmp_mutex   # [linux]
+    - dill
+    - future
+    - gdal
+    - gitpython
+    - matplotlib-base
+    - meshpy
+    - netcdf4
+    - numpy
```
Comment (carterbox) [inline recipes/anuga/meta.yaml:42]:
```suggestion
    - {{ pin_compatible('numpy') }}
```
The numpy version at compile time should match the numpy version at runtime.

**Review by carterbox — CHANGES_REQUESTED** [review]

**Comment by stoiver** [comment]
@conda-forge/help-python-c This recipe should be ready for review.

@carterbox, I changed the numpy requirement to {{ pin_compatible('numpy') }} as suggested. 

I actually went back to the setup.py include requirements and removed most of the specific version constraints (leaving just a constraint on gdal >=3.0.4).

File: recipes/anuga/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  host:
+    - python 
+    - libpython  # [win]
+    - pip
+    - cython
+    - numpy
+
+  run:
+    - python
+    - _openmp_mutex   # [linux]
```
Comment (carterbox) [inline recipes/anuga/meta.yaml:34]:
```suggestion
  build:
    - {{ compiler('c') }}        # [not win]
    - {{ compiler('cxx') }}      # [not win]
    - m2w64-gcc                  # [win]
    - libpython                  # [win]
    - llvm-openmp  # [osx]
    - libgomp      # [linux]

  host:
    - python 
    - libpython  # [win]
    - pip
    - cython
    - numpy

  run:
    - python
```
https://conda-forge.org/docs/maintainer/knowledge_base.html#openmp

File: recipes/anuga/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    - dill
+    - future
+    - gdal >=3.0.4
+    - gitpython
+    - matplotlib-base
+    - meshpy
+    - netcdf4
+    - Pmw
+    - pymetis
+    - pytest
```
Comment (carterbox) [inline recipes/anuga/meta.yaml:45]:
```suggestion
```
Why does your package require pytest at runtime? pytest is ususally for development only.

File: recipes/anuga/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    - numpy
+
+  run:
+    - python
+    - _openmp_mutex   # [linux]
+    - {{ pin_compatible('numpy') }}
+    - dill
+    - future
+    - gdal >=3.0.4
+    - gitpython
```
Comment (carterbox) [inline recipes/anuga/meta.yaml:39]:
```suggestion
```
gitpython also seems like development tool that may not be needed for the main functionality of modeling floods.

File: recipes/anuga/meta.yaml (hunk 4 above)
Comment (carterbox) [reply recipes/anuga/meta.yaml]: Just want to double check: Does your compiled code link against the numpy C API?

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/anuga/meta.yaml (hunk 6 above)
Comment (stoiver) [reply recipes/anuga/meta.yaml:46]: @carterbox, we are using the test  structure provided by numpy which based on pytest. And in addition we use pytest to skip some of those tests. With pytest we can run the tests via  `import anuga; anuga.test()`. So should keep `pytest` in the run requirements.

**Comment by stoiver** [comment]
@conda-forge/help-python-c This recipe should be ready for review.

@carterbox, regarding "Does your compiled code link against the numpy C API?" We use cython to link between our low level C (and in one case C++) code and python. I don't think we now have any explicit use of the numpy C API, other than through the C code created by cython.

**Pull request merged**